idaes/models_extra/co2_capture_and_utilization/unit_models/membrane_1d.py

```diff
@@ -0,0 +1,341 @@
+        "property_package",
+        ConfigValue(
+            default=useDefault,
+            domain=is_physical_parameter_block,
+            description="Property package to use for given stream",
+            doc="""Property parameter object used to define property calculations for given stream,
+    **default** - useDefault.
+    **Valid values:** {
+    **useDefault** - use default package from parent model or flowsheet,
+    **PhysicalParameterObject** - a PhysicalParameterBlock object.}""",
+        ),
+    )
+
+    Stream_Config.declare(
+        "property_package_args",
+        ConfigDict(
+            implicit=True,
+            description="Dict of arguments to use for constructing property package",
+            doc="""A ConfigDict with arguments to be passed to property block(s)
+    and used when constructing these,
+    **default** - None.
+    **Valid values:** {
+    see property package for documentation.}""",
+        ),
+    )
+
+    Stream_Config.declare(
+        "flow_direction",
+        ConfigValue(
+            default=FlowDirection.forward,
+            domain=In(FlowDirection),
+            doc="Direction of flow for stream",
+            description="FlowDirection Enum indicating direction of "
+            "flow for given stream. Default=FlowDirection.forward.",
+        ),
+    )
+
+    Stream_Config.declare(
+        "has_energy_balance",
+        ConfigValue(
+            default=True,
+            domain=Bool,
+            doc="Bool indicating whether to include energy balance for stream. Default=True.",
+        ),
+    )
+
+    Stream_Config.declare(
+        "has_pressure_balance",
+        ConfigValue(
+            default=True,
+            domain=Bool,
+            doc="Bool indicating whether to include pressure balance for stream. Default=True.",
+        ),
+    )
+
+    Stream_Config.declare(
+        "has_feed",
+        ConfigValue(
+            default=True,
+            domain=Bool,
+            doc="Bool indicating whether stream has a feed.",
+            description="Bool indicating whether stream has a feed Port and inlet "
+            "state, or if all flow is provided via mass transfer. Default=True.",
+        ),
+    )
+    CONFIG.declare(
+        "sweep_flow",
+        ConfigValue(
+            default=True,
+            domain=Bool,
+            doc="Bool indicating whether there is a sweep flow in the permeate side.",
+            description="Bool indicating whether stream has a feed Port and inlet "
+            "state, or if all flow is provided via mass transfer. Default=True.",
+        ),
+    )
+
+    CONFIG.declare(
+        "finite_elements",
+        ConfigValue(
+            default=5,
+            domain=int,
+            description="Number of finite elements in length domain",
+            doc="""Number of finite elements to use when discretizing length
+                domain (default=5)""",
+        ),
+    )
+
+    CONFIG.declare(
+        "flow_type",
```

Comment:
This also appears to duplicate options from `StreamCONFIG` (again I suspect you only need this one).

Reply:
I think it's more clear to have "finite_elements" in unit config level.